tests/GenerativeAbstractClassesTests.fs

```diff
@@ -0,0 +1,122 @@
+#if INTERACTIVE
+#load "../src/ProvidedTypes.fsi" "../src/ProvidedTypes.fs" 
+#load "../src/ProvidedTypesTesting.fs"
+
+#else
+
+module FSharp.TypeProviders.SDK.Tests.GenerativeAbstractClassesTests
+#endif
+
+#nowarn "760" // IDisposable needs new
+
+#if !NO_GENERATIVE
+
+open System
+open System.Reflection
+open Microsoft.FSharp.Core.CompilerServices
+open Xunit
+open ProviderImplementation.ProvidedTypes
+open ProviderImplementation.ProvidedTypesTesting
+
+[<TypeProvider>]
+type GenerativeAbstractClassesProvider (config: TypeProviderConfig) as this =
+    inherit TypeProviderForNamespaces (config)
+
+    let ns = "AbstractClasses.Provided"
+    let tempAssembly = ProvidedAssembly()
+    let container = ProvidedTypeDefinition(tempAssembly, ns, "Contracts", Some typeof<obj>, isErased = false)
+
+    let createAbstractClass name (members: (string * (string * Type) list * Type * bool) list) =
+        let t = ProvidedTypeDefinition(name, Some typeof<System.MarshalByRefObject>, hideObjectMethods = true, isErased = false)
+        t.AddAttributes(TypeAttributes.Abstract)
+        
+        members
+        |> List.map (fun (name, parameters, retType, isVirtual) ->
+            let ps =
+                parameters
+                |> List.map (fun (name, ty) ->
+                    ProvidedParameter(name, ty))
+            if isVirtual then
+                let m = ProvidedMethod(name, ps, retType, invokeCode = fun args ->
+                    <@ raise (NotImplementedException(name + " is not implemented")) @>.Raw
+                    )
+                m.AddMethodAttrs (MethodAttributes.Virtual ||| MethodAttributes.HasSecurity)
+                m
+            else
+                let m = ProvidedMethod(name, ps, retType)
+                //m.AddMethodAttrs (MethodAttributes.Virtual ||| MethodAttributes.Abstract)
```

Comment:
Uncommenting this line results in both tests in this file failing in the same way as the interface generation.